docs/source/NRunner.rst

```diff
@@ -0,0 +1,185 @@
+  runner plugins actually execute an Avocado Job remotely, with all
+  the overhead and complications that it brings)
+* Test execution is limited to a single test at a time (non-parallel)
+* Test processes are not properly isolated and can affect the test
+  runner (including the "UI")
+
+And these are some features which it's believed to be more easily
+implemented under a different architecture and implementation:
+
+* Remote execution
+* Different test execution isolation models provided by the test runner
+  (process, container, virtual machine)
+* Distributed execution of tests across a pool of any combination of
+  processes, containers, virtual machines, etc.
+* Parallel execution of tests
+* Optimized runners for a given environment and or test type (for
+  instance, a runner written in RUST to run tests written in RUST
+  in an environment that already has RUST installed but not much
+  else)
+* Notification of execution results to many simultaneous "status
+  servers"
+* Disconnected test execution, so that results can be saved to a
+  device and collected by the runner
+* Simplified and automated deployment of the runner component into
+  execution environments such as containers and virtual machines
+
+Concepts:
+=========
+
+Runnable
+--------
+
+A runnable is a description of an entity that can be executed and
+produce some kind of result.  The description is abstract on purpose.
+A simple and obvious candidate for filling the description is a
+standalone executable, such as the ones available on your ``/bin``
+directory.
+
+A runnable must declare its kind.  Using the previous example of
+standalone executables, those may be given the unique kind identifier
+such as ``exec``.
+
+Each runnable kind may require a different amount of information to be
+provided so that it can be instantiated.  Using standalone executables
+as an example, the information required should be limtied to the
+location of the the standalone executable file.  The following
+pseudo-code may help to put these ideas together::
+
+  runnable_instance = create_runnable('exec', uri='/bin/true')
+
+Runner
+------
+
+A runner is an active entity that acts on the information of a
+runnable.  A runner will usually be tied to an specific kind of
+runnable, and will be able to able to act upon the specific
+information that runnable kind provides.
+
+The following pseudo-code may help to illustrate that::
+
+  runnable_instance = create_runnable('exec', path='/bin/sleep')
+  if runnable.kind == 'exec':
+     runner = create_runner_exec(runnable)
+
+A runner should produce status information on the progress of the
+execution of a runnable.  If the runnable produces interesting
+information, it should forward that along.  For instance, using the
+``exec`` runner example, it's helpful to start producing status
+that the process has been created and it's running as soon as
+possible.  These can be as simple as a sequence of::
+
+  {"status": "RUNNING"}
```

Comment:
Double checking you didn' mean "CREATED".

Reply:
Actually, the exact status have been set in stone.  I've added "INIT" status to the regular Avocado runner, so I wouldn't be surprise if we end up adding something like this here too.